### DISCUSSION TITLE
Import of pdf in cookbook

### DISCUSSION THREAD

**Discussion by quinoa5** [nextcloud/cookbook#1533] [FAQ and user questions]
Is it possible to import pdf files into Nextcloud Cookbook or an just a photo of an e.g. handwritten recipe?

**Comment by christianlupus** [comment]
At the moment this is not possible.

A PDF contains of potentially multiple pages. That means we would have to parse the PDF unless we wanted to store it without any metadata parsing.

Maybe we can support multiple images per recipe (already on the agenda). The OCR (image to text) would still be on the user, though.

**Comment by WhiteTomX** [comment]
Now with the rise of AI, maybe we can leverage that to interpret the photo/pdf and convert them into correct json format? We could ad this as optional feature so people can add common apis for that?

**Comment by officecitek** [comment]
I think it would be sufficient if PDF files or photos could be saved as attachments. This would make it possible to create recipes quickly by simply taking pictures of existing handwritten recipes and adding them as attachments, or adding existing PDF recipes. Later, when there is time, one can manually enter all the necessary data

**Comment by Nicouschulas** [comment]
> Is it possible to import pdf files into Nextcloud Cookbook or an just a photo of an e.g. handwritten recipe?

It's a bit late but if you still need it take a look at this https://github.com/nextcloud/cookbook/discussions/3338
It's working fully automatic, normal text, handwriting etc (currently only image)
